tfjs-backend-webgpu/src/webgpu_program.ts

```diff
@@ -219,16 +224,23 @@ function makeShader(
       }
     `);
 
-  if (program.isFromPixels) {
+  if (program.pixelsOpType != null) {
+    const inoutSnippet = program.pixelsOpType === PixelsOpType.FROM_PIXELS ?
+        `@group(0) @binding(0) var<storage, read_write> result: array<${
+            dataTypeToGPUType(outputData.dtype, program.outputComponent)}>;` :
+        `@group(0) @binding(1) var<storage, read> inBuf : array<${
+            dataTypeToGPUType(inputInfo[0].dtype, program.outputComponent)}>;`;
+    const outShapeStridesType =
+        outputData.shape.length === 3 ? 'vec2<i32>' : 'i32';
     prefixSnippets.push(`
         struct Uniform {
+          outShapeStrides : ${outShapeStridesType},
           size            : i32,
           numChannels     : i32,
-          outShapeStrides : vec2<i32>,
+          alpha           : f32,
```

Comment:
`alpha` is not needed for fromPixels?

Reply:
Yes, put it here to make code look simple

Reply:
Yes, put it here to make code look simple